python/ray/dag/compiled_dag_node.py

```diff
@@ -800,8 +800,17 @@ def _get_or_compile(
                 # in the DAG.
                 readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
 
-                if isinstance(readers[0].dag_node, MultiOutputNode):
-                    assert len(readers) == 1
+                dag_nodes = [reader.dag_node for reader in readers]
```

Comment:
Do you mean the following example is invalid? `a` has two readers, `MultiOutputNode` and `b`.

```python
with InputNode() as inp:
    dag = a.inc.bind(inp)
    branch = b.inc.bind(dag)
    branch = c.inc.bind(branch)
    dag = MultiOutputNode([dag, branch])
```

Reply:
I think my understanding is correct after I read `test_driver_and_actor_as_readers`.

Reply:
Yes, that example should cause an exception to be thrown. The `dag` output is read by both actor `b` and the driver, whereas only the driver (for now) should read the DAG output.